README.md

```diff
@@ -44,16 +45,19 @@ See [INSTALL.md](/INSTALL.md) for detailed installation instructions.
 
 ### Versioning
 
-`client-go` follows [semver](http://semver.org/). We will not make
-backwards-incompatible changes without incrementing the major version number. A
-change is backwards-incompatible either if it *i)* changes the public interfaces
+`client-go` follows [semver](http://semver.org/)-like tags.
```

Comment:
since we don't currently have the ability to increment major versions (xref https://github.com/kubernetes/kubernetes/issues/84372), I would limit this to describing the mapping from Kubernetes release to client-go release